src/components/FeaturePanel/PublicTransport/LineNumber.tsx

```diff
@@ -21,29 +24,44 @@ const LineNumberWrapper = styled.a<{
   display: inline;
 `;
 
-interface LineNumberProps {
-  name: string;
-  color: string;
-  osmType: string;
-  osmId: string | number;
-}
-
-export const LineNumber: React.FC<LineNumberProps> = ({
-  name,
-  color,
-  osmType,
-  osmId,
-}) => {
+type Tags = Record<string, string>;
+const formatTooltip = (tags: Tags, routes: { tags: Tags }[]) => {
+  const formatTags = (tags: Tags) => {
+    const parts = [tags.from, ...(tags.via ? [tags.via] : []), tags.to];
+    return parts.flatMap((str) => str.split(';')).join(' - ');
+  };
```

Comment:
If i understand this correctly, this function is pure, so it can be moved to the top-level. It is always better to use pure functions and also to move them to the top level, because the rest is better readable.

Reply:
Yes it is pure. I absolutely love functional programming and try to do things pure whenever I can.